opentelemetry-user-events-metrics/src/exporter/mod.rs

```diff
@@ -58,17 +64,321 @@ impl Debug for MetricsExporter {
+                                        description: metric.description.clone(),
+                                        unit: metric.unit.clone(),
+                                        data: Box::new(data::ExponentialHistogram {
+                                            temporality: exp_hist.temporality,
+                                            data_points: vec![ExponentialHistogramDataPoint {
+                                                attributes: data_point.attributes.clone(),
+                                                count: data_point.count,
+                                                start_time: data_point.start_time,
+                                                time: data_point.time,
+                                                min: data_point.min,
+                                                max: data_point.max,
+                                                sum: data_point.sum,
+                                                scale: data_point.scale,
+                                                zero_count: data_point.zero_count,
+                                                zero_threshold: data_point.zero_threshold,
+                                                positive_bucket: ExponentialBucket {
+                                                    offset: data_point.positive_bucket.offset,
+                                                    counts: data_point
+                                                        .positive_bucket
+                                                        .counts
+                                                        .clone(),
+                                                },
+                                                negative_bucket: ExponentialBucket {
+                                                    offset: data_point.negative_bucket.offset,
+                                                    counts: data_point
+                                                        .negative_bucket
+                                                        .counts
+                                                        .clone(),
+                                                },
+                                                exemplars: data_point.exemplars.clone(),
+                                            }],
+                                        }),
+                                    }],
+                                }],
+                            };
+                            if let Err(e) = self.serialize_and_write(&resource_metric) {
+                                errors.push(e);
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Return any errors if present
+            if !errors.is_empty() {
+                return Err(MetricsError::Other(format!(
+                    "Encountered {} errors during export",
+                    errors.len()
```

Comment:
This should dump all the errors too, not just the length. Will improve on it in next PR for internal logging/